README.md

```diff
@@ -42,7 +42,7 @@ Celeborn worker's slot count decreases when a partition is allocated and increme
 
 ## Build
 1. Celeborn supports Spark 2.4/3.0/3.1/3.2/3.3/3.4, Flink 1.14/1.15/1.17 and Hadoop MapReduce 2/3.
```

Comment:
As we are here, let's also update the supported Spark versions, Celeborn supports Spark 3.5 since https://github.com/apache/incubator-celeborn/pull/1850. cc @pan3793 @zhouyifan279